rdt-app/src/main/java/io/ona/rdt_app/widget/RDTGpsFactory.java

```diff
@@ -0,0 +1,69 @@
+package io.ona.rdt_app.widget;
+
+import android.content.Context;
+import android.util.DisplayMetrics;
+import android.util.Pair;
+import android.view.View;
+import android.widget.RelativeLayout;
+
+import com.rey.material.widget.Button;
+import com.vijay.jsonwizard.domain.WidgetArgs;
+import com.vijay.jsonwizard.fragments.JsonFormFragment;
+import com.vijay.jsonwizard.interfaces.CommonListener;
+import com.vijay.jsonwizard.widgets.GpsFactory;
+
+import org.json.JSONObject;
+
+import java.util.List;
+
+import io.ona.rdt_app.R;
+import io.ona.rdt_app.fragment.RDTJsonFormFragment;
+import io.ona.rdt_app.util.RDTGpsDialog;
+
+/**
+ * Created by Vincent Karuri on 19/08/2019
+ */
+public class RDTGpsFactory extends GpsFactory {
+
+    private WidgetArgs widgetArgs;
+
+    @Override
+    public List<View> getViewsFromJson(String stepName, final Context context, JsonFormFragment formFragment, JSONObject jsonObject,
+                                       CommonListener listener, boolean popup) throws Exception {
+
+        widgetArgs = new WidgetArgs();
+        widgetArgs.withStepName(stepName)
+                .withContext(context)
+                .withFormFragment(formFragment)
+                .withJsonObject(jsonObject)
+                .withListener(listener)
+                .withPopup(popup);
+
+        List<View> views = super.getViewsFromJson(stepName, context, formFragment, jsonObject, listener, popup);
+        View rootLayout = views.get(0);
+
+        Pair<Integer, Integer> screenDimens = getScreenDimens(context);
+        rootLayout.setLayoutParams(new RelativeLayout.LayoutParams(screenDimens.first, screenDimens.second));
+        rootLayout.findViewById(R.id.altitude).setVisibility(View.GONE);
+        rootLayout.findViewById(R.id.accuracy).setVisibility(View.GONE);
+
+        new RDTJsonFormFragment().setNextButtonState(rootLayout.findViewById(R.id.record_button), true);
+
+        return views;
+    }
+
+    private Pair<Integer, Integer> getScreenDimens(Context context) {
+        DisplayMetrics displayMetrics = context.getResources().getDisplayMetrics();
+        float dpWidth = displayMetrics.widthPixels;
+        float dpHeight = displayMetrics.heightPixels;
+
+        return new Pair<>((int) (dpWidth - (50 * displayMetrics.density)), (int) (dpHeight - (100 * displayMetrics.density)));
```

Comment:
Try to use dimens resources instead of int contants

Reply:
Done.